### PR TITLE
Make `Linear` independent of `Mach`

### PR DESCRIPTION
**Pull request by xclerc**
This pull reqesut makes `Linear` independent
of `Mach`, so that the latter is easier to delete.

The main dependency was the `Lop` constructor,
which is "split" by this PR into:

- `Lop` that will share its nested value with CFG
  (through a new module, namely `Operation`);
- `Lcall_op` that will get the constructors from
  the old `Lop` that cannot be mere operation
  in CFG.

The diff is large, but is mainly about moves and
renames. 

I think it is now handled automatically, but the
magic number for linear should definitely be
bumped.

### CODE REVIEW

File: backend/branch_relaxation.ml [hunk 1]
```diff
@@ -63,11 +63,6 @@ module Make (T : Branch_relaxation_intf.S) = struct
         opt_branch_overflows map pc lbl0 max_branch_offset
           || opt_branch_overflows map pc lbl1 max_branch_offset
           || opt_branch_overflows map pc lbl2 max_branch_offset
-      | Lop (Ipoll { return_label = Some lbl }) ->
-        (* A poll-and-branch instruction can branch to the label lbl,
-           but also to an out-of-line code block. *)
-        code_size + max_out_of_line_code_offset - pc >= max_branch_offset
-        || branch_overflows map pc lbl max_branch_offset
```
Comment (gretay-js) [inline backend/branch_relaxation.ml:70]: `Emit` can still generate these shortcuts, but after this PR Linear cannot represent them. We stopped generating the shortcuts when we removed `Linearize` pass, so this is not changing any existing functionality, but the plan was to recover this optimization on the way out of Cfg. I think these shortcuts were added after some benchmarking upstream, and I hestitate to remove them without a trace. On the other hand, leaving a bunch of dead code in `Emit` is not great. What are you thoughts?

File: backend/amd64/emit.mlp [hunk 2]
```diff
@@ -1762,8 +1753,8 @@ let emit_instr ~first ~fallthrough i =
        intervening wrapper that deals with getting the arguments to the probe
        in the correct place and managing the spilling/reloading of live
        registers.  See [emit_probe_handler_wrapper] below. *)
-    emit_call_probe_handler_wrapper i ~enabled_at_init ~probe_label
-  | Lop (Iprobe_is_enabled { name; }) ->
+                                    emit_call_probe_handler_wrapper i ~enabled_at_init ~probe_label
```
Comment (gretay-js) [inline backend/amd64/emit.mlp:1756]: extra spaces?

**Review by gretay-js — APPROVED** [review]
Not much is left in `cfg_to_linear_desc.ml`, do we still need it to be a separate file?

File: backend/amd64/emit.mlp (hunk 2 above)
Comment (xclerc) [reply backend/amd64/emit.mlp:1756]: Indeed - sorry.

File: backend/branch_relaxation.ml (hunk 1 above)
Comment (gretay-js) [reply backend/branch_relaxation.ml:70]: Actually, `Emit` cannot generate these shortcuts any more after this PR, so I think a comment about it on https://github.com/ocaml-flambda/flambda-backend/issues/2421 would be enough.

**Pull request merged**